contracts/TxRelay.sol

```diff
@@ -10,6 +10,8 @@ contract TxRelay {
     // Different from the nonce defined w/in protocol.
     mapping(address => uint) nonce;
 
+    mapping(address => mapping(address => bool)) public whitelist;
```

Comment:
I think this should have a more descriptive name, or an explanatory comment.

Reply:
Do you think this comment is sufficient?
```
This mapping specifies a whitelist of allowed senders of transactions.
There can be one whitelist per ethereum account. Users can specify
which whitelist they want to use when signing the transaction. They
can also skip using a whitelist by giving the zero address.
```

Reply:
Small changes to explicitly define more of the behavior:

```
This mapping specifies a whitelist of allowed senders for transactions.
There can be one whitelist per ethereum account, which is the owner of that whitelist. 
Users can specify which whitelist they want to use when signing a transaction. They can use their own whitelist, a whitelist belonging to another account, or skip using a whitelist by specifying the zero address.```